scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/ClientHello.java

```diff
@@ -148,16 +154,20 @@ private ClientHello(ProtocolVersion version, SecureRandom secureRandom, SessionI
 		} else {
 			this.sessionId = SessionId.emptySessionId();
 		}
+		if (cipherSuites != null) {
+			this.cipherSuites.addAll(cipherSuites);
+		}
 
-		// the supported groups
-		// TODO make list of supported groups configurable
-		SupportedGroup[] supportedGroups = SupportedGroup.getPreferredGroups().toArray(new SupportedGroup[]{});
-		this.extensions.addExtension(new SupportedEllipticCurvesExtension(supportedGroups));
-
-		// the supported point formats
-		List<ECPointFormat> formats = Arrays.asList(ECPointFormat.UNCOMPRESSED);
-		this.extensions.addExtension(new SupportedPointFormatsExtension(formats));
+		if (CipherSuite.containsEccBasedCipherSuite(cipherSuites)) {
```

Comment:
In this place, the check for the support of an ECC based cipher suite is exactly the right one FMPOV :+1: